### PR TITLE
routing change

### PR DESCRIPTION
**Pull request by DanoFecko**
@dBucik requested changes in routing
'auth' was added to the path
login and preauth components added, for getting user from backend and redirecting

image

### CODE REVIEW

**Comment by DanoFecko** [comment]
@Vojtech-Sassmann, please do the review of this PR

File: gui/src/app/login/login.component.ts [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
-export class LoginComponent {
+export class LoginComponent implements OnInit {
 
-  constructor() { }
+  constructor(private userService: UsersService,
+              private router: Router) {
+  }
+
+  ngOnInit(): void {
+    if (this.userService.getUser() != null) {
```
Comment (Vojtech-Sassmann) [inline gui/src/app/login/login.component.ts]: This is not working as expected. The method `getUser()` returns an Observable object, not the user! You have to subscribe to the object and validate the returned result.

File: gui/src/app/login/login.component.ts [hunk 2]
```diff
@@ -1,12 +1,26 @@
...
+      this.router.navigate(['/auth']);
+    } else {
+      if (this.userService.setUser()) {
```
Comment (Vojtech-Sassmann) [inline gui/src/app/login/login.component.ts]: Same problem here.

File: gui/src/app/login/preauth/preauth.component.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+})
+export class PreauthComponent implements OnInit {
+
+  constructor(private userService: UsersService,
+              private router: Router) {
+  }
+
+  ngOnInit() {
+    console.log("preauth");
+    if (this.userService.getUser() != null) {
```
Comment (Vojtech-Sassmann) [inline gui/src/app/login/preauth/preauth.component.ts]: Same here.

File: gui/src/app/login/preauth/preauth.component.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      this.router.navigate(['/auth']);
+    } else {
+      if (this.userService.setUser()) {
```
Comment (Vojtech-Sassmann) [inline gui/src/app/login/preauth/preauth.component.ts]: Same here.

**Comment by DanoFecko** [comment]
@Vojtech-Sassmann fixed as instructed

**Comment by DanoFecko** [comment]
I found bug in chained routing, fixing it right now

**Comment by DanoFecko** [comment]
bug is fixed

**Comment by DanoFecko** [comment]
i think a fixed it

**Comment by DanoFecko** [comment]
@Vojtech-Sassmann after the review, could you merge it into the other branch (than master), so others can test it thoroughly?

File: gui/src/app/requests/request-detail/request-detail.component.html [hunk 5]
```diff
@@ -63,7 +63,7 @@ <h4 class="mt-4">{{'REQUESTS.REQUEST_DETAIL_ITEMS' | translate}}</h4>
             <div *ngIf="isUndefined(item.value); else notNullValue" class="font-italic text-muted">
               {{'UNDEFINED' | translate}}
             </div>
-            <ng-template #notNullValue>
+            <ng-template #notNullValue>/auth
```
Comment (Vojtech-Sassmann) [inline gui/src/app/requests/request-detail/request-detail.component.html]: This probably shouldn't be here,

Comment (DanoFecko) [reply gui/src/app/requests/request-detail/request-detail.component.html]: removed

**Comment by DanoFecko** [comment]
I created second PR to the branch "routing", for testing purposes

File: gui/src/app/app.component.html [hunk 6]
```diff
@@ -29,7 +29,7 @@
           person
         </mat-icon>
         <div>
-          {{user.fullName}}
+          {{getUser().fullName}}
```
Comment (Vojtech-Sassmann) [inline gui/src/app/app.component.html]: Could you please add verification of the user not being undefined above to get rid of the warnings in console?
`<div class="user-info">` 
=>
`<div class="user-info" *ngIf="getUser() !== undefined">`

Comment (DanoFecko) [reply gui/src/app/app.component.html]: fixed

**Comment by dBucik** [comment]
Merged from separate branch

**Pull request closed without merging**